pkg/machines/components/networks/createNetworkDialog.jsx

```diff
@@ -397,11 +397,17 @@ class CreateNetworkModal extends React.Component {
             this.setState({ inProgress: false, validate: true });
         } else {
             const {
-                connectionName, name, forwardMode, ipv4, ipv6, prefix, device,
+                connectionName, name, forwardMode, ip, prefix, device,
                 ipv4DhcpRangeStart, ipv4DhcpRangeEnd, ipv6DhcpRangeStart, ipv6DhcpRangeEnd
             } = this.state;
             const netmask = utils.netmaskConvert(this.state.netmask);
 
+            let { ipv4, ipv6 } = this.state;
+            if (ip === "IPv4 only")
+                ipv6 = undefined;
```

Comment:
This is a state object, you can't assign to it directly, you need to use setState. I am not sure about the possible side effects tbh though.

Reply:
Oh right. I use this value only to be passed to other function, it's not meant to change state. After this code, the modal gets closed. But If there would be some error during creation, then it could cause problems.